### PR TITLE
fix maprotation condition

### PR DESCRIPTION
**Pull request by Gireen**
possible fix for #1142

### CODE REVIEW

File: src/sgame/sg_maprotation.cpp [hunk 1]
```diff
@@ -1042,6 +1042,14 @@ Send commands to the server to actually change the map
...
 {
 	mrNode_t *node = mapRotations.rotations[ rotation ].nodes[ index ];
+  mrCondition_t *condition;
+
+  if( node->type == NT_CONDITION )
+  {
+    condition = &node->u.condition;
+    node = condition->target;
+  }
+
```
Comment (illwieckz) [inline src/sgame/sg_maprotation.cpp]: interesting, last time I tried to only remove the `|| conditonal` test I got random ugly crashes, maybe I was missing this.

Comment (Gireen) [reply src/sgame/sg_maprotation.cpp]: yeah the rotation will try to load the node with the condition as map, which results in random letters as mapname

**Comment by illwieckz** [comment]
on that topic (but that would be another PR), I would like to be able to ship multiple layout in a map dpk archive, and be able to load them. Even being able to override the builtin bsp layout (in that case, asking for builtin would load the one named default in dpk), that would make entity editing and buildable placement an easy job: just create a map with almost nothing, load in engine, make a layout, dump the layout, store it in dpk and voilà.

**Comment by Gireen** [comment]
that is already mostly possible, layouts can be shipped with maps (see parpax mini)
and with #1144 you can set the default layout. What would be needed is a way to distinguish between builtin-default and server default

**Comment by illwieckz** [comment]
hmm, I wanted to write this comment in the #1144 thread :D

File: src/sgame/sg_maprotation.cpp (hunk 1 above)
Comment (slipher) [inline src/sgame/sg_maprotation.cpp]: some mixed tabs/spaces indentation or something here

File: src/sgame/sg_maprotation.cpp [hunk 2]
```diff
@@ -1042,6 +1042,14 @@ Send commands to the server to actually change the map
 static void G_IssueMapChange( int index, int rotation )
 {
 	mrNode_t *node = mapRotations.rotations[ rotation ].nodes[ index ];
+  mrCondition_t *condition;
```
Comment (slipher) [inline src/sgame/sg_maprotation.cpp]: Prefer to declare variables in the narrowest possible scope (inside the `if` block in this case)

**Comment by illwieckz** [comment]
My server is currently using this code and it works, and I see no unattended side effect.

See #1146 for an issue that make this code not really useful: `numClients` currently count bots…

**Pull request merged**